.github/workflows/build.yaml

```diff
@@ -13,15 +13,20 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v3
         with:
-          java-version: 11
+          distribution: 'zulu'
+          java-version: |
+            11
+            17
+            21
```

Comment:
I added separate test tasks that run on Java 11 and Java 17 to ensure compatibility (https://jakewharton.com/build-on-latest-java-test-through-lowest-java/), so I'm pre-installed all required versions. I skipped applying https://github.com/gradle/foojay-toolchains to let gradle download required versions automatically, I can do that if preferred

Reply:
Ah, I see this also produces a build warning in clear environment 😞 

>Using a toolchain installed via auto-provisioning, but having no toolchain repositories configured. This behavior is deprecated. Consider defining toolchain download repositories, otherwise the build might fail in clean environments; see https://docs.gradle.org/8.4/userguide/toolchains.html#sub:download_repositories

I probably should add the foojay toolchains plugin 👀